src/main/java/com/redhat/devtools/lsp4ij/features/completion/LSPCompletionProposal.java

```diff
@@ -316,6 +337,12 @@ protected void apply(Document document, char trigger, int stateMask, int offset)
             }
 
             List<TextEdit> additionalEdits = item.getAdditionalTextEdits();
+            if (additionalEdits == null && supportResolveCompletion) {
+                CompletionItem resolved = getResolvedCompletionItem();
```

Comment:
maybe we should impose a timeout to the resolve request, to prevent it from blocking completion

Reply:
This pattern is used on every features and cancel the future when:

 * the Psifile changed
 * there are a ProgressManager is canceled.

For the moment I have never seen problem, so I suggest that we keep like this and see if there are some problem with some language server.